selfhost/codegen.jakt

```diff
@@ -2562,7 +2377,11 @@ struct CodeGenerator {
         }
         output += "debug_description() const {\n"
         output += "auto builder = DeprecatedStringBuilder::create();\n"
-        output += "switch (this->index()) {"
+        output += "switch (this->__jakt_init_index()) {"
+
+        // NOTE: This should be replaced by a set of strings and checking each
+        // if `StructLike`'s fields aren't guaranteed to contain all of the common fields first.
```

Comment:
wdym by this?
common fields are only guaranteed to be _present_ in all variants, the placement can be arbitrary if it helps you out.

Reply:
By "common" fields I mean the fields that are declared in `enum_.fields`, which are duplicated only in `StructLike`'s fields. Those are found always at the start of `StructLike`'s fields, although I can switch to using a `{String}` to avoid putting a constraint there.

Reply:
That is: ```jakt
enum SomeEnum {
    common_value: String
    VariantA(span: Span)
    VariantB(i32)
}
``` has `common_value` in both `enum_.fields` and `VariantA`'s `CheckedVariant::StructLike`

Reply:
ah, _that_'s what you mean.

Reply:
should be fine to assume they always come first, the codegen implementation is the only component that cares.

Reply:
OK, so I can keep this then. Should I add a similar note (or a redirection to this one) to the rest of the places where I use this assumption? It's not the only place that has `let common_field_count = enum_.fields.size()`

Reply:
Up to you really.
one should be enough for a "the order matters" warning imo.

Reply:
I'll move the note to the type definition then.